graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py

```diff
@@ -1579,25 +1584,55 @@ def write_edge_structure(
         edge_df_with_int_ids = edge_df_with_int_ids.drop(src_col, dst_col).repartition(
             self.num_output_files
         )
-        edge_df_with_int_ids_and_all_features = edge_df_with_int_ids
-        edge_df_with_only_int_ids = edge_df_with_int_ids.select(["src_int_id", "dst_int_id"])
 
         edge_structure_path = os.path.join(
             self.output_prefix, f"edges/{edge_type.replace(':', '_')}"
         )
-        logging.info("Writing edge structure for edge type %s...", edge_type)
-        if self.add_reverse_edges:
+
+        edge_df_with_int_ids_and_all_features = edge_df_with_int_ids
+        edge_df_with_only_int_ids = edge_df_with_int_ids.select(["src_int_id", "dst_int_id"])
+        if self.add_reverse_edges and not self.is_homogeneous:
             edge_df_with_only_int_ids.cache()
-        path_list = self._write_df(edge_df_with_only_int_ids, edge_structure_path)
 
-        if self.add_reverse_edges:
+        if self.add_reverse_edges and not self.is_homogeneous:
```

Comment:
See comment on previous version, should be easier to read as two nested if loops instead of using the `and`